pkg/ingester/ingester_v2_test.go

```diff
@@ -1815,6 +1815,48 @@ func TestIngester_flushing(t *testing.T) {
 				m.AssertNumberOfCalls(t, "Sync", 1)
 			},
 		},
+
+		"noBlockSpanningOverADayOnFlush": {
+			setupIngester: func(cfg *Config) {
+				cfg.BlocksStorageConfig.TSDB.FlushBlocksOnShutdown = false
+			},
+
+			action: func(t *testing.T, i *Ingester, m *shipperMock) {
+				// Pushing 5 samples, spanning over 3 days.
+				pushSingleSampleAtTime(t, i, 23*time.Hour.Milliseconds())
+				pushSingleSampleAtTime(t, i, 24*time.Hour.Milliseconds()-1)
+				pushSingleSampleAtTime(t, i, 24*time.Hour.Milliseconds()+1)
+				pushSingleSampleAtTime(t, i, 25*time.Hour.Milliseconds())
+				pushSingleSampleAtTime(t, i, 26*time.Hour.Milliseconds())
+
+				// Nothing shipped yet.
+				m.AssertNumberOfCalls(t, "Sync", 0)
+
+				i.FlushHandler(httptest.NewRecorder(), httptest.NewRequest("POST", "/flush", nil))
+
+				// Flush handler only triggers compactions, but doesn't wait for them to finish. Let's wait for a moment, and then verify.
+				time.Sleep(3 * time.Second)
```

Comment:
Instead of `time.Sleep()` you could use `util.Poll()` checking the number of times `Sync()` has been called. This would make test faster and avoid flaky test if it takes more than 3s for any reason.